src/ast/irbuilderbpf.cpp

```diff
@@ -299,6 +299,8 @@ llvm::Type *IRBuilderBPF::GetType(const SizedType &stype)
     ty = getInt64Ty();
   } else if (stype.IsVoidTy()) {
     ty = getVoidTy();
+  } else if (stype.IsBoolTy()) {
+    ty = getInt1Ty();
```

Comment:
What's the purpose of this, given that you're overriding it with `getInt8Ty()` above and below in this PR?

It would be nice if we could just use this `getInt1Ty()` and not have the overrides though. I'm surprised that LLVM doesn't make this work. Maybe it's just a problem with the BPF backend that converts `i1` into `i8`? (I'm just guessing here)

Reply:
The `getInt8Ty()` override is only necessary when doing proberead as it requires to read the entire byte. For all other purposes, this `getInt1Ty()` should do the job.

But I agree that the code was confusing with the `getInt8Ty()` overrides spread all over the place so I moved everything related to the proberead special case to one place.